pkgs/development/python-modules/pyscard/ignore-macos-bug.patch

```diff
@@ -1,19 +1,18 @@
 diff --git a/test/test_SCardGetErrorMessage.py b/test/test_SCardGetErrorMessage.py
-old mode 100644
-new mode 100755
-index c6fe755..c1217f5
+index ea4a167..c1217f5 100644
 --- a/test/test_SCardGetErrorMessage.py
 +++ b/test/test_SCardGetErrorMessage.py
-@@ -29,12 +29,10 @@ class TestError(unittest.TestCase):
-         self.assertEqual(res, expected)
- 
+@@ -31,14 +31,8 @@ class TestError(unittest.TestCase):
          res = SCardGetErrorMessage(1)
-+        expected = "Unknown error: 0x00000001"
+         expected = "Unknown error: 0x00000001"
          # macOS bug not yet fixed
--        if get_platform().startswith('macosx-') and get_platform() < 'macosx-10.13':
--            expected = "Unkown error: 0x00000001"
--        else:
--            expected = "Unknown error: 0x00000001"
+-        if get_platform().startswith('macosx-'):
+-            version = get_platform()  # something like 'macosx-10.14-x86_64'
+-            version = version.split('-')[1]  # '10.14'
+-            major, minor = map(int, version.split('.'))  # (10, 14)
+-            if major == 10 and minor < 13:
+-                expected = "Unkown error: 0x00000001"
+-
 -        self.assertEqual(res, expected)
 +        macos_bug_expected = "Unkown error: 0x00000001"
 +        self.assertIn(res, [expected, macos_bug_expected])
```

Comment:
Looks good. Should we just propose this change upstream?

Reply:
I was about to make the same comment. This isn't nix-specific, so it should probably be upstreamed.

Reply:
I agree with that. I'll make a pull request and propose the change.

Reply:
This has been fixed in https://github.com/LudovicRousseau/pyscard/pull/96

I'm not too familiar with the Python ecosystem, but my sense in this will take a release to get in. Would it make sense to ship this patch for nixpkgs to get to 1.9.9, and then drop it when it proves no longer necessary in the next release?